Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,45 @@ def installed?
       !versions.empty?
     end
 
+    def outdated?(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
+      !@outdated[greedy].empty?
+    end
+
+    def outdated_versions(greedy = false)
+      greedy = true if greedy
+      @outdated ||= {}
+      @outdated[greedy] ||= _check_outdated(greedy)
```

Comment:
Let's not cache this. At the moment we are only using it once anyways, but even if we were to use it multiple times, something will probably have changed between these calls.